mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLHostOps.td

```diff
@@ -110,4 +110,33 @@ def SYCLHostHandlerSetNDRange : SYCL_HostOp<"handler.set_nd_range"> {
   let hasVerifier = 1;
 }
 
+def SYCLHostLaunchKernel : SYCL_HostOp<"launch_kernel",
+    [DeclareOpInterfaceMethods<SymbolUserOpInterface>,
+     AttrSizedOperandSegments]> {
+  let summary = "Launches a SYCL kernel.";
+  let description = [{
+    This operation represents the launch of a kernel function with the given
+    range and kernel arguments.
```

Comment:
Can we document the limitations/things that were left out from this operation (modelling of queue, events) here?

Reply:
I am wondering whether the existing gpu.launch or similar function could be sufficient. 
I am not opposed to add a custom operation in the SYCL dialect, if that is convenient though.

Reply:
I think it's convenient for us to refer to the SYCL-typed ranges directly. `gpu.launch_func` expects this info broken down into 6 scalars.

Reply:
Another possible reason to avoid using gpu.launch could be to prevent unwanted cross host device optimizations, as our host code is going to be removed.